lib/platform.c

```diff
@@ -627,21 +628,35 @@ void STDCALL sf_os_version(char *ret, size_t size) {
         sb_strcpy(ret, size, envbuf.release);
     }
 #elif _WIN32
-    // https://msdn.microsoft.com/en-us/library/windows/desktop/ms724833%28v=vs.85%29.aspx
-    // Version  Actual Windows Version
-    //    10.0  Windows 10.0/Server 2016
-    //     6.3  Windows  8.1/Server 2012 R2
-    //     6.2  Windows  8.0/Server 2012 R1
-    //     6.1  Windows  7.0/Server 2008 R2
-    //     6.0  Windows  Vist/Server 2008 R1 (Not supported)
-    OSVERSIONINFOEX info;
-    ZeroMemory(&info, sizeof(OSVERSIONINFOEX));
-    info.dwOSVersionInfoSize = sizeof(OSVERSIONINFOEX);
-    GetVersionEx((LPOSVERSIONINFO)&info);
+    int majorVersion = 0;
+    int minorVersion = 0;
+
+    if (IsWindows10OrGreater())
```

Comment:
our jenkins builds report problem with this function on VS14:
```
2024-02-07 03:35:57 snowflakeclient.lib(platform.obj) : error LNK2019: unresolved external symbol _IsWindows10OrGreater referenced in function _sf_os_version@8
```

Reply:
@sfc-gh-dprzybysz 
It should be fixed with my changes.
I can't reproduce it on my local though since I've had vs2022 installed and the windows sdk updated to 10.x.
Please rerun the build on jenkins to ensure the issue is solved. It would take some effort for me to get a clear environment that  only has Windows sdk 8.1 available.

Reply:
The build was successful on jenkins